logger_test.go

```diff
@@ -62,43 +61,43 @@ func stubExit() *stubbedExit {
 	return stub
 }
 
-func withJSONLogger(t testing.TB, opts []Option, f func(*jsonLogger, func() []string)) {
+func withJSONLogger(t testing.TB, opts []Option, f func(Logger, *testBuffer)) {
 	sink := newTestBuffer()
 	errSink := newTestBuffer()
 
 	allOpts := make([]Option, 0, 3+len(opts))
 	allOpts = append(allOpts, DebugLevel, Output(sink), ErrorOutput(errSink))
 	allOpts = append(allOpts, opts...)
-	jl := NewJSON(allOpts...)
-	jl.StubTime()
+	logger := newLogger(newJSONEncoder(NoTime()), allOpts...)
 
-	f(jl.(*jsonLogger), func() []string {
-		output := strings.Split(sink.String(), "\n")
-		return output[:len(output)-1]
-	})
-	assert.Empty(t, errSink.String(), "Expected error sink to be empty")
+	f(logger, sink)
+	assert.Empty(t, errSink.Lines(), "Expected error sink to be empty")
```

Comment:
when checking if it's empty, you can probably just use `errSink.String()`, no point trying to convert to lines right?